utility_scripts/quota_audit.py

```diff
@@ -0,0 +1,171 @@
+# Copyright 2017 Massachusetts Open Cloud
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+"""Audits quota requests
+Utility script for finding projects that don't match quota requests.
+Usage:
+    python quota_audit.py
+"""
+import os
+import csv
+from keystoneclient.v3 import client
+from keystoneauth1 import session
+from keystoneauth1.identity import v2
+from keystoneauth1.exceptions.http import NotFound
+from neutronclient.v2_0 import client as nclient
+from novaclient import client as novaclient
+from cinderclient.v2 import client as cinderclient
+
+# Use only one of the auth sections below
+
+"""
+# Uncomment this section to authenticate with a config file
+#CONFIG_FILE = '/path/to/file'
+#config = ConfigParser.ConfigParser()
+#config.read(CONFIG_FILE)
+#admin_user = config.get('auth', 'admin_user')
+#admin_pwd = config.get('auth', 'admin_pwd')
+#admin_project = config.get('auth', 'admin_project')
+#auth_url = config.get('auth', 'auth_url')
+"""
+
+# Uncomment this section to authenticate with an environment set from a
+# keystonerc file
```

Comment:
The other scripts like this only have the singular way of authenticating, not 3 different ways